go/cmd/vtgate/vtgate.go

```diff
@@ -49,6 +49,74 @@ func init() {
 	servenv.RegisterDefaultFlags()
 }
 
+// FindVal takes a slice and looks for an element in it. If found it will
+// return it's key, otherwise it will return -1 and a bool of false.
+func FindVal(slice []string, val string) (int, bool) {
+	for i, item := range slice {
+		if item == val {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
+// CheckCellFlags will check validation of cell and cells_to_watch flag
+// it will help to avoid strange behaviors when vtgate runs but actually does not work
+func CheckCellFlags(ctx context.Context, serv srvtopo.Server, cell string, cellsToWatch string) error {
+	//topo check
+	var topoServer *topo.Server
+	if serv != nil {
+		var err error
+		topoServer, err = serv.GetTopoServer()
+		if err != nil {
+			log.Exitf("Unable to create gateway: %v", err)
+		}
+	} else {
+		log.Exitf("topo server can not be nil")
+	}
+	cellsInTopo, err := topoServer.GetKnownCells(ctx)
+	if err != nil {
+		return err
+	}
+	if len(cellsInTopo) == 0 {
+		return fmt.Errorf("topo server should at least has one cell")
+	}
+
+	//cell valid check
+	if cell == "" || cell == "test_nj" {
+		return fmt.Errorf("cell flag must be set")
+	}
+	hasCell := false
+	for _, v := range cellsInTopo {
+		if v == cell {
+			hasCell = true
+		}
+	}
+	if !hasCell {
+		return fmt.Errorf("cell:[%v] does not exist in topo", cell)
+	}
+
+	//cells_to_watch valid check
+	cells := make([]string, 0, 1)
+	for _, c := range strings.Split(cellsToWatch, ",") {
+		if c == "" {
+			continue
+		}
+		//cell should be contained in cellsInTopo
+		_, exists := FindVal(cellsInTopo, c)
```

Comment:
@enisoc suggests using `topo.InCellList` can be used instead of writing another function.

Reply:
see, will fix